Site Management Panel: Update primary styling on buttons

DustyReagan opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/8783

## Proposed Changes

* This PR removes the "primary" button styling from the buttons in the Site Management flyout panels.



Before | After
--|--
<img width="1728" alt="Screenshot 2024-09-04 at 10 15 23 AM" src="https://github.com/user-attachments/assets/61a66e51-3498-4eb9-9585-fa0656d0d0a1">  | <img width="1726" alt="Screenshot 2024-09-04 at 10 15 05 AM" src="https://github.com/user-attachments/assets/c1d9fc59-7926-4820-adcf-bcfe708e09a6">


Before | After
--|--
<img width="1727" alt="Screenshot 2024-09-04 at 10 15 34 AM" src="https://github.com/user-attachments/assets/3916ddca-0979-49d9-8303-af640909dc76"> | <img width="1728" alt="Screenshot 2024-09-04 at 10 15 44 AM" src="https://github.com/user-attachments/assets/18c0d70d-0b6e-47f4-aadf-cb51b19d573b">

Before | After
--|--
<img width="1728" alt="Screenshot 2024-09-04 at 10 16 04 AM" src="https://github.com/user-attachments/assets/bfa0e156-226a-46d9-be4c-673ce1ac7c36"> | <img width="1728" alt="Screenshot 2024-09-04 at 10 15 55 AM" src="https://github.com/user-attachments/assets/781eeb97-fd9e-4e5a-9b5c-99337650852d">

Before | After
--|--
<img width="1728" alt="Screenshot 2024-09-04 at 10 37 39 AM" src="https://github.com/user-attachments/assets/28ad0ed7-2786-4d0a-93ea-e8f150c5c05f">  | <img width="1728" alt="Screenshot 2024-09-04 at 10 37 28 AM" src="https://github.com/user-attachments/assets/7601091b-99cd-4db6-b4ed-3e3f409ea194">





Before | After
--|--
<img width="1728" alt="Screenshot 2024-09-04 at 10 16 19 AM" src="https://github.com/user-attachments/assets/03efed25-4cd9-4c93-b29a-9c88fbb3ddd8"> | <img width="1728" alt="Screenshot 2024-09-04 at 10 16 29 AM" src="https://github.com/user-attachments/assets/cf1d7bae-eeca-4fa5-80f1-fdf0d0fb577f">

Before | After
--|--
<img width="1728" alt="Screenshot 2024-09-04 at 10 45 40 AM" src="https://github.com/user-attachments/assets/18b6b876-0156-40ac-be6f-37093893aeb1">  | <img width="1728" alt="Screenshot 2024-09-04 at 10 46 07 AM" src="https://github.com/user-attachments/assets/05544db5-fcb3-4a4d-b2a7-ae700945a9cb">



## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Decrease competition for the user's attention with too many call-to-action styled buttons.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Use Calypso Live
* Go to /sites and check all of Site Management tabs for "primary" buttons. They should be "regular" buttons, like in the screenshots above.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

DustyReagan committed:
Remove primary styling from buttons

DustyReagan committed:
Update staging site buttons

DustyReagan committed:
Update create credentials button

allilevine commented:
This could be a personal preference, but I think the toggles or "segmented controls" look better with a background. It's hard to tell what's selected at a glance.

<img width="303" alt="Screen Shot 2024-09-04 at 4 26 01 PM" src="https://github.com/user-attachments/assets/5a60719a-db41-42ac-a1cc-d7a8da61cf30">
<img width="466" alt="Screen Shot 2024-09-04 at 4 25 55 PM" src="https://github.com/user-attachments/assets/117c93ba-6809-4676-9649-59d7ce50e099">

The pagination is highlighted with a background:

<img width="320" alt="Screen Shot 2024-09-04 at 4 26 40 PM" src="https://github.com/user-attachments/assets/3a6a1513-e864-4a97-914f-3a62fe85ccd8">


I found another primary button on a staging site > Staging site tab, should that be a non-primary button too?

<img width="1458" alt="Screen Shot 2024-09-04 at 4 27 09 PM" src="https://github.com/user-attachments/assets/f4b971c4-9cf7-49bc-ac4c-4d5c58ae6fe8">

DustyReagan committed:
Remove primary style from switch to primary site button

DustyReagan committed:
Use gray background for toggle and page number

DustyReagan commented:
Thanks, @allilevine! From this conversation p1725482783281379-slack-C03NLNTPZ2T I've updated the background color of the page numbers and the toggles and updated the "Switch to production site" button.
<img width="291" alt="Screenshot 2024-09-04 at 4 59 54 PM" src="https://github.com/user-attachments/assets/a438e5e3-0812-4650-a243-cef28b9a7c5d">
<img width="466" alt="Screenshot 2024-09-04 at 4 59 49 PM" src="https://github.com/user-attachments/assets/4bf704a4-a9e3-4c44-8f91-eaba0ad708c8">

davemart-in commented:
Some feedback:

The add site button should probably have a secondary button style:

![CleanShot 2024-09-05 at 07 15 50@2x](https://github.com/user-attachments/assets/99958268-a7b3-4575-bd02-f7a848b211ba)

I think it's right to have the WP Admin button the primary button on the Overview page:

![CleanShot 2024-09-05 at 07 20 40@2x](https://github.com/user-attachments/assets/560a1583-70c2-4a66-bb96-f455386fb8f8)

But in subsequent tabs, I could see us making it the secondary button and allowing the primary button to be used elsewhere. Here are a few examples:

1)Hosting features tab for simple sites:

![CleanShot 2024-09-05 at 07 22 50@2x](https://github.com/user-attachments/assets/21176d87-17de-4057-8d32-171c5ea6c168)

2) Deployments:

![CleanShot 2024-09-05 at 07 24 19@2x](https://github.com/user-attachments/assets/b7042fd8-f6fe-4cc3-bcfc-550625f7b668)

3) Logs:

![CleanShot 2024-09-05 at 07 25 09@2x](https://github.com/user-attachments/assets/87b197e5-b100-47c5-a0b9-6ce948f5966a)

4) Staging site:

![CleanShot 2024-09-05 at 07 25 52@2x](https://github.com/user-attachments/assets/e663fbc0-3d4b-4363-8e0b-8c33a763c8a1)

5) The server settings tab is a tough one. Realistically, these should all be on their own separate page, but I think we explored that option and decided not to go that route. 

This almost looks like a text field instead of a button:

![CleanShot 2024-09-05 at 07 27 43@2x](https://github.com/user-attachments/assets/df3c677a-473b-4e86-8943-d031af35b55b)

Can we change the button copy on this to be just "Restore". That should make this feel more like a button.

Then I might just make the Open PHPMyAdmin button the primary button.

DustyReagan committed:
Make new site button secondary when preview pane is open

DustyReagan committed:
Make home button primary on hosting overview only

DustyReagan committed:
Update the primary buttons on each tab

DustyReagan committed:
Update restore button copy

DustyReagan committed:
Make sectionName optional and fix type error' did not match any files

DustyReagan commented:
Thanks, @davemart-in! I've updated the PR with your feedback. It's ready for another look.

davemart-in reviewed (approved):
Thanks for working through all of this! These changes are a good more in the right direction.
